packages/mui-material/src/LinearProgress/LinearProgress.js

```diff
@@ -228,36 +295,64 @@ const LinearProgressBar2 = styled('span', {
       ownerState.variant === 'buffer' && styles.bar2Buffer,
     ];
   },
-})(
-  ({ ownerState, theme }) => ({
-    width: '100%',
-    position: 'absolute',
-    left: 0,
-    bottom: 0,
-    top: 0,
-    transition: 'transform 0.2s linear',
-    transformOrigin: 'left',
-    ...(ownerState.variant !== 'buffer' && {
-      backgroundColor:
-        ownerState.color === 'inherit'
-          ? 'currentColor'
-          : (theme.vars || theme).palette[ownerState.color].main,
-    }),
-    ...(ownerState.color === 'inherit' && {
-      opacity: 0.3,
-    }),
-    ...(ownerState.variant === 'buffer' && {
-      backgroundColor: getColorShade(theme, ownerState.color),
-      transition: `transform .${TRANSITION_DURATION}s linear`,
-    }),
-  }),
-  ({ ownerState }) =>
-    (ownerState.variant === 'indeterminate' || ownerState.variant === 'query') &&
-    css`
-      width: auto;
-      animation: ${indeterminate2Keyframe} 2.1s cubic-bezier(0.165, 0.84, 0.44, 1) 1.15s infinite;
-    `,
-);
+})(({ theme }) => ({
+  width: '100%',
+  position: 'absolute',
+  left: 0,
+  bottom: 0,
+  top: 0,
+  transition: 'transform 0.2s linear',
+  transformOrigin: 'left',
+  variants: [
+    ...Object.entries(theme.palette)
+      .filter(([, value]) => value.main)
+      .map(([color]) => ({
+        props: { color },
+        style: {
+          '--LinearProgressBar2-barColor': (theme.vars || theme).palette[color].main,
+        },
+      })),
+    {
+      props: ({ ownerState }) => ownerState.variant !== 'buffer' && ownerState.color !== 'inherit',
+      style: {
+        backgroundColor: 'var(--LinearProgressBar2-barColor, currentColor)',
+      },
+    },
+    {
+      props: {
+        color: 'inherit',
+      },
+      style: {
+        opacity: 0.3,
+      },
+    },
+    ...Object.entries(theme.palette)
+      .filter(([, value]) => value.main)
+      .map(([color]) => ({
+        props: { color, variant: 'buffer' },
+        style: {
+          backgroundColor: getColorShade(theme, color),
+          transition: `transform .${TRANSITION_DURATION}s linear`,
+        },
+      })),
+    {
+      props: ({ ownerState }) =>
+        ownerState.variant === 'indeterminate' || ownerState.variant === 'query',
+      style: {
+        width: 'auto',
+      },
+    },
+    ...(typeof indeterminate2Animation !== 'string'
+      ? [
+          {
+            props: ({ ownerState }) =>
+              ownerState.variant === 'indeterminate' || ownerState.variant === 'query',
+            style: indeterminate2Animation,
```

Comment:
Does this actually work on Pigment side ?

Reply:
Yes, it works.